Resources/public/js/deprecated/Admin.js

```diff
@@ -0,0 +1,280 @@
+                        $(navbar).width('auto');
+                    } else {
+                        $(navbar).width($(wrapper).outerWidth());
+                    }
+                }
+            });
+        }
+
+        if (footer.length) {
+            new Waypoint({
+                element: wrapper[0],
+                offset: 'bottom-in-view',
+                handler: function (direction) {
+                    var position = $('.sonata-ba-form form > .row').outerHeight() + $(footer).outerHeight() - 2;
+
+                    if (position < $(footer).offset().top) {
+                        $(footer).removeClass('stuck');
+                    }
+
+                    if (direction == 'up') {
+                        $(footer).addClass('stuck');
+                    }
+                }
+            });
+        }
+
+        handleScroll(footer, navbar, wrapper);
+    }
+
+    function handleScroll (footer, navbar, wrapper) {
+        if (footer.length && $(window).scrollTop() + $(window).height() != $(document).height()) {
+            $(footer).addClass('stuck');
+        }
+
+        $(window).on('scroll', debounce(function() {
+            if (footer.length && $(window).scrollTop() + $(window).height() == $(document).height()) {
+                $(footer).removeClass('stuck');
+            }
+
+            if (navbar.length && $(window).scrollTop() === 0) {
+                $(navbar).removeClass('stuck');
+            }
+        }, 250));
+
+        $('body').on('expanded.pushMenu collapsed.pushMenu', function() {
+            handleResize(footer, navbar, wrapper);
+        });
+
+        $(window).on('resize', debounce(function() {
+            handleResize(footer, navbar, wrapper);
+        }, 250));
+    }
+
+    function handleResize (footer, navbar, wrapper) {
+        setTimeout(function() {
+            if (navbar.length && $(navbar).hasClass('stuck')) {
+                $(navbar).width($(wrapper).outerWidth());
+            }
+
+            if (footer.length && $(footer).hasClass('stuck')) {
+                $(footer).width($(wrapper).outerWidth());
+            }
+        }, 350); // the animation take 0.3s to execute, so we have to take the width, just after the animation ended
+    }
+
+    $(function() {
+        setupStickyElements($(document));
+    });
+
+    //
+    // Formerly Admin.set_object_field_value()
+    // This seems to be the legacy xeditable implementation, and is not currently used anywhere, AFAICT.
+    // In this case, it should be removed
+    // ---------------------------------------------------------------------------------------------------------------
+
+    function setObjectFieldValue ($subject) {
+        $.ajax({
+            url: $subject.attr('href'),
+            method: 'post',
+            dataType: 'json'
+        }).done(function (json) {
+            var $parent = $subject.parent();
+            if (json.status === "OK") {
+                // fix issue with html comment ...
```

Comment:
That's why I hate "fix bug" comments or commit messages… it makes the code sacred without letting people question if there is indeed a bug and if this is the best way to solve it. Which leads to this : ![genius or stupid](http://www.commitstrip.com/wp-content/uploads/2016/01/Strip-Un-code-inutile-650-finalenglish-1.jpg)